### PR TITLE
update to 2023.4.0

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
fastparquet update to 2023.4.0
- add pinnings to host section
- skip s390x because lack of cramjam & numba
- move numpy from run_constrained to run as it is in the requirements https://github.com/dask/fastparquet/blob/f747fe606e3b7c0c8c23f07d169296c862d69a54/requirements.txt#LL2C1-L2C1
- added cramjam dependency
- added description
- updated doc_url

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,67 +1,67 @@
...
     - pip
-    - pytest-runner
+    - packaging
+    - wheel
     - setuptools
-    - thrift >=0.11
+    - numpy 
+    - setuptools-scm 7.0.4
+    - cython 0.29.32
+    - pandas 1.5.2
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:33]: `pandas` and `packaging` aren't required for `host`, see https://github.com/dask/fastparquet/blob/c8028e00de2e5cb8594c442d2c1837019a61553a/pyproject.toml#L2 or https://github.com/dask/fastparquet/blob/c8028e00de2e5cb8594c442d2c1837019a61553a/setup.py#L56-L61

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,67 +1,67 @@
...
+    - pytest-runner 6.0.0
   run:
     - python
-    - numba >=0.49
-    - packaging
-    - pandas >=1.1.0
+    - {{ pin_compatible('pandas') }}
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:37]: `pandas >=1.5.0` is enough

**Comment by skupr-anaconda** [comment]
Could you try this https://github.com/conda-forge/fastparquet-feedstock/blob/57c46efc32c513b8c679c48b9214d2751ab0d845/recipe/meta.yaml#L16-L18

**Comment by skupr-anaconda** [comment]
Could you try add this https://github.com/conda-forge/fastparquet-feedstock/blob/57c46efc32c513b8c679c48b9214d2751ab0d845/recipe/meta.yaml#LL18C6-L18C6 + file https://github.com/conda-forge/fastparquet-feedstock/blob/main/recipe/un_oldest.py

File: recipe/meta.yaml (hunk 2 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: done

File: recipe/meta.yaml (hunk 1 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: done

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,67 +1,66 @@
...
-  # There is a missing numba package.
-  skip: True  # [py<36 or s390x]
+  number: 0
+  #skip s390x because of lack of cramjam and numba
+  skip: true  # [py<38 or s390x]
   script:
     - del fastparquet\*.c  # [win]
     - rm fastparquet/*.c   # [unix]
-    - {{ PYTHON }} -m pip install . --no-deps -vv
+    - "{{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv "
```
Comment (boldorider4) [inline recipe/meta.yaml:19]: are double quotes necessary?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,67 +1,66 @@
...
   requires:
     - pip
+  commands:
+    - pip check
 
 about:
   home: https://github.com/dask/fastparquet
   license: Apache-2.0
   license_family: Apache
   license_file: LICENSE
```
Comment (boldorider4) [inline recipe/meta.yaml:54]: shouldn't it be LICENSE.txt?

File: recipe/meta.yaml (hunk 3 above)
Comment (Jrice1317) [inline recipe/meta.yaml:19]:
```suggestion
    - {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv 
```

File: recipe/meta.yaml (hunk 4 above)
Comment (Jrice1317) [reply recipe/meta.yaml:54]: No, I think we're okay to leave it as is. See the path [here](https://github.com/dask/fastparquet/blob/2023.4.0/LICENSE): https://github.com/dask/fastparquet/blob/2023.4.0/LICENSE

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,67 +1,66 @@
...
   summary: 'Python interface to the parquet format'
+  description: |
+    The fastparquet is a python implementation of the parquet format, aiming integrate into python-based big data work-flows. 
+    It is used implicitly by the projects Dask, Pandas and intake-parquet. 
+    The Parquet format is a common binary data store, used particularly in the Hadoop/big-data sphere. 
+    It provides several advantages relevant to big-data processing.
+
   dev_url: https://github.com/dask/fastparquet
-  doc_url: https://github.com/dask/fastparquet/blob/main/docs/source/quickstart.rst
+  doc_url: https://fastparquet.readthedocs.io/en/latest/
```
Comment (Jrice1317) [inline recipe/meta.yaml:63]:
```suggestion
  doc_url: https://fastparquet.readthedocs.io/
```

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,67 +1,66 @@
...
-    - numba
-    - numpy
-    - packaging
-    - pandas
     - pip
-    - pytest-runner
+    - wheel
     - setuptools
-    - thrift >=0.11
+    - numpy
```
Comment (Jrice1317) [inline recipe/meta.yaml:30]: @skupr-anaconda Should we be pinning `numpy` here according to [this](https://github.com/dask/fastparquet/blob/2023.4.0/setup.py#L61)?

Comment (skupr-anaconda) [reply recipe/meta.yaml:30]: Yeah, that makes sense. `numpy {{ numpy }}` would be better for `host`

File: recipe/meta.yaml [hunk 7]
```diff
@@ -1,67 +1,66 @@
...
-    - numpy
-    - packaging
-    - pandas
     - pip
-    - pytest-runner
+    - wheel
     - setuptools
-    - thrift >=0.11
+    - numpy 
+    - setuptools-scm 7.0.4
```
[the remark this answers is not in the document]
Comment (skupr-anaconda) [reply recipe/meta.yaml:31]: Not sure in this case. v7.0.4 is the first that supports python 3.11 so here it's correct

File: recipe/meta.yaml (hunk 5 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: done

File: recipe/meta.yaml (hunk 3 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: removed

Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: removed

**Review by ELundby45 — APPROVED** [review]
LGTM

**Pull request merged**